[ServiceLocatorToDIRector] Error "Unable to resolve repository argument"

# Bug Report

When running Rector with set `doctrine-repository-as-service`, I get the following error : 

```                                                                                           
 [ERROR] Could not process "tests/Unit/App/Doctrine/ItemRepositoryTest.php" file by  
         "Rector\Rector\Architecture\RepositoryAsService\ServiceLocatorToDIRector", due to:                             
         "Unable to resolve repository argument".                                                                       
```

## Minimal PHP Code Causing Issue

https://getrector.org/demo/c301546c-7d92-4b99-b1b6-1955831a6f39

## Expected Behaviour

No errors ? ^^

--- a/rules/architecture/src/Rector/MethodCall/ServiceLocatorToDIRector.php
+++ b/rules/architecture/src/Rector/MethodCall/ServiceLocatorToDIRector.php
@@ -8,6 +8,7 @@ use Nette\Utils\Strings;
 use PhpParser\Node;
 use PhpParser\Node\Expr\ClassConstFetch;
 use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
 use PhpParser\Node\Name;
 use PhpParser\Node\Scalar\String_;
 use PhpParser\Node\Stmt\Class_;
@@ -22,7 +23,7 @@ use Rector\Doctrine\Contract\Mapper\DoctrineEntityAndRepositoryMapperInterface;
 use Rector\NodeTypeResolver\Node\AttributeKey;
 
 /**
- * @see \Rector\Core\Tests\Rector\Architecture\DoctrineRepositoryAsService\DoctrineRepositoryAsServiceTest
+ * @see \Rector\Architecture\Tests\Rector\DoctrineRepositoryAsService\DoctrineRepositoryAsServiceTest
  */
 final class ServiceLocatorToDIRector extends AbstractRector
 {
@@ -100,6 +101,12 @@ PHP
     public function refactor(Node $node): ?Node
     {
         if (! $this->isName($node->name, 'getRepository')) {
+            return null;
+        }
+
+        $firstArgumentValue = $node->args[0]->value;
+        // possible mocking → skip
+        if ($firstArgumentValue instanceof StaticCall) {
             return null;
         }
 

--- a/rules/architecture/tests/Rector/DoctrineRepositoryAsService/Fixture/issues_2930.php.inc
+++ b/rules/architecture/tests/Rector/DoctrineRepositoryAsService/Fixture/issues_2930.php.inc
@@ -1,0 +1,17 @@
+<?php
+
+namespace Rector\Architecture\Tests\Rector\DoctrineRepositoryAsService\Fixture;
+
+use Doctrine\ORM\EntityManagerInterface;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+
+final class ItemRepositoryTest extends TestCase
+{
+    public function testGetThrowsExceptionIfNotFound(): void
+    {
+        $entityManagerMock = $this->prophesize(EntityManagerInterface::class);
+
+        $entityManagerMock->getRepository(Argument::any());
+    }
+}

